src/Aspire.Hosting/ApplicationModel/ResourceExtensions.cs

```diff
@@ -571,6 +571,40 @@ public static int GetReplicaCount(this IResource resource)
         }
     }
 
+    /// <summary>
+    /// Gets the deployment target for the specified resource, if any. Throws an exception if
+    /// there are multiple compute environments and a compute environment is not explicitly specified.
+    /// </summary>
+    public static DeploymentTargetAnnotation? GetDeploymentTargetAnnotation(this IResource resource)
+    {
+#pragma warning disable ASPIRECOMPUTE001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+        if (resource.TryGetLastAnnotation<ComputeEnvironmentAnnotation>(out var computeEnvironmentAnnotation))
+        {
+            // find the deployment target for the compute environment
+            return resource.Annotations
+                .OfType<DeploymentTargetAnnotation>()
+                .LastOrDefault(a => a.ComputeEnvironment == computeEnvironmentAnnotation.ComputeEnvironment);
+        }
+#pragma warning restore ASPIRECOMPUTE001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+        else
+        {
+            DeploymentTargetAnnotation? result = null;
+            foreach (var annotation in resource.Annotations.OfType<DeploymentTargetAnnotation>())
```

Comment:
Make the error show the compute environment names in the error when there's more than one?

Reply:
Done.